src/utils.ts

```diff
@@ -256,15 +256,21 @@ export function calculateEditorColFromVimScreenCol(
     let currentCharIdx = 0;
     let currentVimCol = 0;
     while (currentVimCol < screenCol) {
-        const bytes =
-            line[currentCharIdx] === "\t"
-                ? tabSize - (currentVimCol % tabSize)
-                : useBytes
-                ? getBytesFromCodePoint(line.codePointAt(currentCharIdx))
-                : wcwidth(line[currentCharIdx]);
-
-        currentVimCol += bytes;
-        currentCharIdx += bytes === 4 ? 2 : 1;
+        if (line[currentCharIdx] === "\t") {
+            currentVimCol += tabSize - (currentVimCol % tabSize);
+            currentCharIdx++;
+        }
+        else if (useBytes) {
+            const bytes = getBytesFromCodePoint(line.codePointAt(currentCharIdx));
+            currentVimCol += bytes;
+            // Characters which take 4 bytes also take 2 string indices.
+            // (Only relevant here since wcwidth returns half of the value to each part)
+            currentCharIdx += bytes === 4 ? 2 : 1;
```

Comment:
I understand why this check is relevant only here, and therefore feel comfortable with limiting it here, but I still don't understand this function - why do we want to handle tabs differently? they count as just one column when using `col`.
Also, I don't get _why_ wcwidth is used, but I do understand that it doesn't suffer from the problem emoji had